OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -839,7 +839,8 @@ public void ResolveOrder(Actor self, Order order)
 			else if (order.OrderString == "ReturnToBase" && rearmableInfo != null && rearmableInfo.RearmActors.Any())
 			{
 				UnReserve();
-				self.CancelActivity();
+				if (!order.Queued)
+					self.CancelActivity();
```

Comment:
Minor style nit: Please insert an empty line below this, to make it more visible that this 'if' has nothing to do with the one below.

Reply:
fixed